### DISCUSSION TITLE
hb_buffer_destroy

### DISCUSSION THREAD

**Discussion by ProdigyView** [harfbuzz/harfbuzz#4444] [Q&A] [answered]
I am having issues compiling ffmpeg with libfreetype with this commmand on windows MSYS2:

    ./configure     --pkg-config-flags="--static" --enable-libvpl  --enable-libopenh264     
    --enable-version3  --enable-libfreetype    --enable-libopus     --enable-libvpx     --enable-libvorbis     
    --enable-libaom     --enable-libdav1d     --disable-gpl     --disable-w32threads     --enable-pthreads     
    --disable-shared     --enable-static     
    --extra-cflags='--static'   --extra-cflags="-I/mingw64/include -static"   --extra-ldflags="-L/mingw64/lib -static" --prefix="/home/compiled"

I keep getting this error:

    ERROR: freetype2 not found using pkg-config

    If you think configure made a mistake, make sure you are using the latest
    version from Git.  If the latest version fails, report the problem to the
    ffmpeg-user@ffmpeg.org mailing list or IRC #ffmpeg on irc.libera.chat.
    Include the log file "ffbuild/config.log" produced by configure as this will help
    solve the problem.

I've installed freetype with this command:

     pacman -Ss mingw-w64-x86_64-freetype
    
I've also tried compiling freetype2 from the source:

    git clone https://git.savannah.gnu.org/git/freetype/freetype2.git
    
    cd freetype2
    
    mkdir build && cd build
    
    /mingw64/bin/cmake .. -G "MSYS Makefiles" -DBUILD_SHARED_LIBS=OFF -DCMAKE_INSTALL_PREFIX=/mingw64
    
    make
    
    make install

And pkg-config gives this:

    pkg-config freetype2 --cflags --libs
    -IC:/msys64/mingw64/include/freetype2 -IC:/msys64/mingw64/include/harfbuzz -IC:/msys64/mingw64/include/glib-2.0 -IC:/msys64/mingw64/lib/glib-2.0/include -IC:/msys64/mingw64/include/libpng16 -lfreetype

I can confirm the package is there:

    ls /mingw64/lib/pkgconfig/freetype2.pc
    /mingw64/lib/pkgconfig/freetype2.pc

My logs are giving this erro:

    C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/13.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: 
    C:/msys64/mingw64/lib\libfreetype.a(src_autofit_autofit.c.obj):(.text+0x7cfc): undefined reference to `hb_buffer_destroy'
    collect2.exe: error: ld returned 1 exit status
    ERROR: freetype2 not found using pkg-config

Which is this library?

Is there anything else I should be doing to compile a static version of ffmpeg with this package?

**Comment by behdad** [comment]
What does `pkg-config freetype2 --cflags --libs --static` give?

**Reply by ProdigyView** [reply]
@behdad Here is what I get:

```
pkg-config freetype2 --cflags --libs --static
-IC:/msys64/mingw64/include/freetype2 -IC:/msys64/mingw64/include/libpng16 -IC:/msys64/mingw64/include/harfbuzz -IC:/msys64/mingw64/include/glib-2.0 -IC:/msys64/mingw64/lib/glib-2.0/include -L/mingw64/lib -lbz2 -lpng16 -lharfbuzz -lm -lbrotlidec -lbrotlicommon -lglib-2.0 -lintl -lws2_32 -lole32 -lwinmm -lshlwapi -luuid -lm -lpcre2-8 -lfreetype -lz
```

**Reply by behdad** [reply]
That looks right to me. Sorry, I'm not familiar with ffmpeg build system enough to know what you are doing wrong.

**Comment by khaledhosny** [comment, marked as answer]
This is a question for ffmpeg maintainers, not HarfBuzz.
